packages/patternfly-3/react-console/src/ConsoleSelector/ConsoleSelector.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Grid, Row, Col, Form, FormGroup, Dropdown, MenuItem } from 'patternfly-react';
+import { Checkbox } from 'react-bootstrap';
+
+import { NONE_TYPE, SERIALCONSOLE_TYPE, VNCCONSOLE_TYPE } from './constants';
+
+class ConsoleSelector extends React.Component {
+  state = {
+    type: NONE_TYPE,
+    disconnectByChange: this.props.disconnectByChange,
+    keptConnection: {} // no connection exists when mounted
+  };
+
+  onTypeChange(type) {
+    this.setState(prevState => {
+      const keptConnection = prevState.disconnectByChange
+        ? { [type]: true }
+        : Object.assign({}, prevState.keptConnection, { [type]: true });
```

Comment:
can use the [spread operator?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)

Reply:
sure, done